core/src/main/java/com/linecorp/armeria/client/http/HttpClientPipelineConfigurator.java

```diff
@@ -114,20 +116,13 @@
     private InetSocketAddress remoteAddress;
 
     HttpClientPipelineConfigurator(SessionProtocol sessionProtocol, SessionOptions options) {
-        switch (sessionProtocol) {
-        case HTTP:
-        case HTTPS:
+        if (sessionProtocol == HTTP || sessionProtocol == HTTPS) {
             httpPreference = HttpPreference.HTTP2_PREFERRED;
-            break;
-        case H1:
-        case H1C:
+        } else if (sessionProtocol == H1 || sessionProtocol == H1C) {
             httpPreference = HttpPreference.HTTP1_REQUIRED;
-            break;
-        case H2:
-        case H2C:
+        } else if (sessionProtocol == H2 || sessionProtocol == H2C) {
             httpPreference = HttpPreference.HTTP2_REQUIRED;
-            break;
-        default:
+        } else {
```

Comment:
Anyways, I agree that we will need more advanced mechanism if we want to allow the modification of the existing protocol implementation's behavior. However, I'm not even sure how complicated it will be or if it's just better forking the original protocol implementation and advertising it as a completely different protocol - It's not easy to tell without adding such a new protocol actually.